core/src/main/java/net/sourceforge/jnlp/runtime/classloader/DelegatingClassLoader.java

```diff
@@ -0,0 +1,59 @@
+package net.sourceforge.jnlp.runtime.classloader;
+
+import java.net.URL;
+import java.util.Objects;
+
+public class DelegatingClassLoader extends ClassLoader {
+    public static final DelegatingClassLoader instance = new DelegatingClassLoader(Thread.currentThread().getContextClassLoader());
+    private ClassLoader classLoader;
+
+    public static DelegatingClassLoader getInstance() {
+        return instance;
+    }
+
+    DelegatingClassLoader(ClassLoader loader) {
+        super(loader);
+        this.classLoader = loader;
+    }
+
+    public void setClassLoader(ClassLoader loader) {
+        this.classLoader = loader;
+    }
+
+    protected Class findClass(String name) throws ClassNotFoundException {
+        return this.classLoader.loadClass(name);
+    }
+
+    protected URL findResource(String name) {
+        return this.classLoader.getResource(name);
+    }
+
+    public int hashCode() {
+        int result = 1;
+        result = 31 * result + (this.classLoader == null ? 0 : this.classLoader.hashCode());
+        result = 31 * result + (this.getParent() == null ? 0 : this.getParent().hashCode());
+        return result;
+    }
+
+    public boolean equals(Object obj) {
```

Comment:
Since this is a singleton we simply check do `return this == obj` ;)

Reply:
You are right we could do it that way. However not sure if i would recommend it. Do you want me to change it?

Reply:
Adding @sclassen to have an additional opinion.

Reply:
Just leave the method as it is. Then it is safe for future refactoring